app/scripts/background.js

```diff
@@ -327,6 +327,27 @@ function maybeDetectPhishing(theController) {
   );
 }
 
+/**
+ * Overrides the Content-Security-Policy Header, acting as a workaround for an MV2 Firefox Bug.
+ */
+function overrideContentSecurityPolicyHeader() {
+  browser.webRequest.onHeadersReceived.addListener(
+    ({ responseHeaders }) => {
+      for (const header of responseHeaders) {
+        if (header.name.toLowerCase() === 'content-security-policy') {
+          header.value = header.value.replace(
+            /(^|;\s*)script-src([^;]*)/u,
+            (match) => `${match} 'nonce-${btoa(browser.runtime.getURL('/'))}'`,
+          );
+        }
+      }
+      return { responseHeaders };
+    },
+    { urls: ['http://*/*', 'https://*/*'] },
+    ['blocking', 'responseHeaders'],
```

Comment:
We probably want to limit the matching to: `types: ['main_frame', 'sub_frame'],`. Can you think of any reasons we'd want to match any and all requests?

Reply:
You should use `urls` and `types`.